### ISSUE TITLE
[Feature Request] Multiple DbContext Support

### ISSUE THREAD

**Issue by theprateik** [ErikEJ/EFCorePowerTools#83]
Recently, I ran into a requirement of referencing multiple databases in a single project. 
Latest version of ``EFCorePowerTools`` can generate multiple ``DatabaseContexts``, but overrides the configuration in ``efpt.config.json`` with the last used ``DatabaseContext``.

Is there any plan on supporting configuration for multiple ``datacontexts`` in the ``efpt.config.json``?

**Comment by ErikEJ** [comment]
Not currently.. I am aware of the issue, and this is the first time it has come up as an issue.

**Comment by theprateik** [comment]
Ok. Thanks.
Not saying I will submit a PR, but, are you willing to accept/look into PR relating to this?

**Comment by ErikEJ** [comment]
Absolutely - maybe something like efpt.MYContext.config.json ?

**Comment by theprateik** [comment]
That sounds fine. 
But, I was thinking more like saving configuration as array in the current ``efpt.config.json``?
Do you think separate config file is better than having array in single file?

**Comment by ErikEJ** [comment]
Yes, that is probably even better, the real challenge is how to connect a config section with a database connection (Context name is not the key here, database connection is)

**Comment by theprateik** [comment]
Yes, that ran into my mind as well. 

Correct my understanding here.

What is wrong with assigning Context name as the key? Each Context name can only point to a single database connection. Hence, if the user chooses (for whatever reason), user can even have multiple Database Contexts pointing to single database connection as long as they keep the Context names unique.

If that doesn't make sense, how about Database name as the key. It should be unique to the database connection, right?

**Comment by ErikEJ** [comment]
The issue is that the config file also shows information about the selected tables, and that is before the ContextName is in play - so maybe just database name should be key!

**Comment by theprateik** [comment]
As I think of database name,  I think can have conflicts, because multiple server/instances can have the databases with same Database name.
I am still think Context name is better choice for the key. What if a temporary (unique) context name is generated and used until the user gets to choose their context name?

**Comment by ErikEJ** [comment]
If you have multiple contexts in the same name, do you think it is likely that they will refer to the same database name? I am referring to only the name of the User Database.

**Comment by theprateik** [comment]
I am not quite sure what you mean in your last comment. But, I think I got the idea why Context Name cannot be used as key here. 

User selects the database, selects the tables and then user gets to choose the Context Name. 
At this point, the application doesn't know how to link the user selections to existing Context Names (if there are any). 

I see your point. Thanks.

**Comment by ErikEJ** [comment]
@LouisGordon Why not have each context in its own project? Seems to me like an anti-pattern to have multiple in same project.

**Comment by ErikEJ** [comment]
@theprateik @LouisGordon @aetos382 @blogcraft 

The latest daily build allows you to use multiple efpt.config.json files in the same project - https://github.com/ErikEJ/EFCorePowerTools/wiki/Reverse-Engineering#saving-options-and-running-the-second-time

**Comment by ErikEJ** [comment]
@gemolnar It can be the name of your database, or the name of your DbContext, you decide, as long as the file follow the efpt.*.config.json naming pattern, it will be recognized.

**Comment by ErikEJ** [comment]
@gemolnar Docs are here: https://github.com/ErikEJ/EFCorePowerTools/wiki/Reverse-Engineering#saving-options-and-running-the-second-time - appreciate any feedback.

**Comment by gemolnar** [comment]
I upgraded to the latest daily build (2.4.242) and now it works for me. I created multiple `efpt.xyz.config.json` files and when I click `Reverse Engineer` it lets me choose which one to use.
Thanks!

**Comment by ErikEJ** [comment]
@gemolnar thanks for confirming, yes, as you can see from the Release notes, this is currently only available in the daily build.
